lib/ansible/modules/clustering/kubevirt/kubevirt_cdi_upload.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2018, KubeVirt Team <@kubevirt>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+
+module: kubevirt_cdi_upload
+
+short_description: Upload local VM images to CDI Upload Proxy.
+
+version_added: "2.8"
+
+author: KubeVirt Team (@kubevirt)
+
+
+description:
+  - Use Openshift Python SDK to create UploadTokenRequest objects.
+  - Transfer contents of local files to the CDI Upload Proxy.
+
+extends_documentation_fragment:
+  - k8s_auth_options
+
+options:
+  pvc_name:
+    description:
+      - Use to specify the name of the target PersistentVolumeClaim.
+    required: true
+  pvc_namespace:
+    description:
+      - Use to specify the namespace of the target PersistentVolumeClaim.
+    required: true
+  upload_host:
+    description:
+      - URL containing the host and port on which the CDI Upload Proxy is available.
+  upload_host_verify_ssl:
+    description:
+      - Whether or not to verify the CDI Upload Proxy's SSL certificates.
+    default: true
+    type: bool
+  path:
+    description:
+      - Path of local image file to transfer.
+  kind:
+    description:
+      - Required for compatibility with ansible <2.8. Can't be modified.
+    default: UploadTokenRequest
+    choices:
+      - UploadTokenRequest
+  api_version:
+    description:
+      - Required for compatibility with ansible <2.8. Can't be modified.
+    default: upload.cdi.kubevirt.io/v1alpha1
+    choices:
+      - upload.cdi.kubevirt.io/v1alpha1
+  src:
+    description:
+      - Required for compatibility with ansible <2.8. Don't use.
+  resource_definition:
+    description:
+      - Required for compatibility with ansible <2.8. Don't use.
+  merge_type:
+    description:
+      - Required for compatibility with ansible <2.8. Don't use.
+  name:
+    description:
+      - Required for compatibility with ansible <2.8. Don't use.
+  namespace:
+    description:
+      - Required for compatibility with ansible <2.8. Don't use.
+
+
+requirements:
+  - python >= 2.7
+  - openshift >= 0.6.2
+  - requests >= 2.0.0
+'''
+
+EXAMPLES = '''
+- name: Upload local image to pvc-vm1
+  kubevirt_cdi_upload:
+    pvc_namespace: default
+    pvc_name: pvc-vm1
+    upload_host: https://localhost:8443
+    upload_host_verify_ssl: false
+    path: /tmp/cirros-0.4.0-x86_64-disk.img
+'''
+
+RETURN = '''# '''
```

Comment:
So you think we shouldn't return anything, only `change=true/false`?

Reply:
Initially I assumed we'd be returning something interesting. But it's a file upload module, there's really no useful data that it can return for later processing.